sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero_command_buffer.cpp

```diff
@@ -558,6 +560,20 @@ UR_APIEXPORT ur_result_t UR_APICALL urCommandBufferEnqueueExp(
     ur_exp_command_buffer_handle_t hCommandBuffer, ur_queue_handle_t hQueue,
     uint32_t numEventsInWaitList, const ur_event_handle_t *phEventWaitList,
     ur_event_handle_t *phEvent) {
+
+  // Submit dependent open command lists for execution, if any
+  for (uint32_t I = 0; I < numEventsInWaitList; I++) {
+    ur_event_handle_t_ *Event =
+        ur_cast<ur_event_handle_t_ *>(phEventWaitList[I]);
+    auto UrQueue = Event->UrQueue;
+    if (UrQueue) {
+      // Lock automatically releases when this goes out of scope.
+      std::scoped_lock<ur_shared_mutex> lock(UrQueue->Mutex);
+
+      UR_CALL(UrQueue->executeAllOpenCommandLists());
```

Comment:
Is this call blocking (could you put that in a code comment too)? i.e does it just submit the work in the queue or does it also wait for it to finish?

Reply:
I've removed this code as it wasn't actually doing anything for this issue, sorry about that!